pretyping/globEnv.ml

```diff
@@ -161,21 +161,21 @@ let interp_ltac_variable ?loc typing_fun env sigma id : Evd.evar_map * unsafe_ju
        inside the try but I want to avoid refactoring this function
        too much for now. *)
     typing_fun {env with lvar; static_env = env.renamed_env} term
-  with Not_found ->
+  | None ->
   (* Check if [id] is a ltac variable not bound to a term *)
   (* and build a nice error message *)
-  if Id.Map.mem id env.lvar.ltac_genargs then begin
-    let Geninterp.Val.Dyn (typ, _) = Id.Map.find id env.lvar.ltac_genargs in
-    user_err ?loc
-     (str "Variable " ++ Id.print id ++ str " should be bound to a term but is \
-      bound to a " ++ Geninterp.Val.pr typ ++ str ".")
-  end;
   if Id.Map.mem id env.lvar.ltac_idents then begin
     let bnd = Id.Map.find id env.lvar.ltac_idents in
     user_err ?loc
      (str "Variable " ++ Id.print id ++ str " should be bound to a term but is \
       bound to the identifier " ++ quote (Id.print bnd) ++ str ".")
   end;
+  if Id.Map.mem id env.lvar.ltac_genargs then begin
```

Comment:
why the permutation of checks btw?

Reply:
The previous order meant the second check was a noop, because (the domain of) ltac_idents is always a subset of ltac_genargs.
Alternatively we could delete the ltac_idents check since it seems nobody cared that it did nothing.

Reply:
It just gives a more precise message. I'm not sure it's really that useful, but in any case this should be for another PR.